src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV1.cs

```diff
@@ -55,19 +56,127 @@ public class DownloadPipelineArtifactTaskV1 : PipelineArtifactTaskPluginBaseV1
+                    else
+                    {
+                        // This should not happen since the build id comes from build environment. But a user may override that so we must be careful.
+                        throw new ArgumentException(StringUtil.Loc("BuildIdIsNotValid", buildIdStr));
+                    }
+                } 
+                //Create the directory if absent.
+                string fullPath = Path.GetFullPath(downloadPath);
+                bool isDir = Directory.Exists(fullPath);
+                if (!isDir)
+                {
+                    Directory.CreateDirectory(fullPath);
+                }
+
+                context.Output(StringUtil.Loc("DownloadArtifactTo", downloadPath));
+                PipelineArtifactServer server = new PipelineArtifactServer();
+                await server.DownloadAsyncMinimatch(context, projectId, buildIdInt, artifactName, downloadPath, minimatchPatterns, token);
+                context.Output(StringUtil.Loc("DownloadArtifactFinished"));
+            }
+            else if(buildType == "specific")
+            {
+                // Project Name - coming from the GUI.
+                int buildIdInt = 0;
+                if(buildVersionToDownload == "latest")
+                {
+                    List<int> defns= new List<int>();
+                    defns.Add(Int32.Parse(buildPipelineDefinition));
+                    VssConnection connection = context.VssConnection;
+                    BuildHttpClient _buildHttpClient = connection.GetClient<BuildHttpClient>();
+                    List<Build> list = await _buildHttpClient.GetBuildsAsync(project, definitions: defns, tagFilters: tagsInput,  queryOrder: BuildQueryOrder.FinishTimeDescending);
+                    if (list.Any())
+                    {
+                        buildIdInt = list.First().Id;
+                    }       
+                }
+                else if(buildVersionToDownload == "specific")
+                {
+                    buildIdInt = Int32.Parse(buildId);
+                }
+                else if(buildVersionToDownload == "latestFromBranch")
+                {
+                    List<int> defns = new List<int>();
+                    defns.Add(Int32.Parse(buildPipelineDefinition));
+                    VssConnection connection = context.VssConnection;
+                    BuildHttpClient _buildHttpClient = connection.GetClient<BuildHttpClient>();
+                    List<Build> list = await _buildHttpClient.GetBuildsAsync(project, definitions: defns, branchName: branchName, tagFilters: tagsInput, queryOrder: BuildQueryOrder.FinishTimeDescending);
+                    if (list.Any())
+                    {
+                        buildIdInt = list.First().Id;
+                    }
+                }
+                //Create the directory if absent.
+                string fullPath = Path.GetFullPath(downloadPath);
+                bool isDir = Directory.Exists(fullPath);
```

Comment:
create common function for directory creation as it is use dmore than once

Reply:
Moved to a common function.